ethereumj-core/src/main/java/org/ethereum/config/CommonConfig.java

```diff
@@ -201,7 +201,7 @@ public void fastSyncCleanUp() {
 
         EthereumListener.SyncState syncStage = EthereumListener.SyncState.values()[fastsyncStageBytes[0]];
 
-        if (!systemProperties().isFastSyncEnabled() || syncStage == EthereumListener.SyncState.UNSECURE) {
+        if ((systemProperties().isSyncEnabled() && !systemProperties().isFastSyncEnabled()) || syncStage == EthereumListener.SyncState.UNSECURE) {
```

Comment:
It's better to add this check to the beginning of the method cause whole method call doesn't make sense if sync is OFF